### DISCUSSION TITLE
NovoPen scanning inconsistent uploading to Nightscout

### DISCUSSION THREAD

**Discussion by rossiscool123** [NightscoutFoundation/xDrip#3470] [Q&A]
### Your environment
- What version of xDrip: 3e86af7-2023.11.14
- What type of hardware data source and how do you get your data there: Dex Share Follower

### Expected behavior
All treatments scanned from the NovoPen should be uploaded from xDrip to nightscout via the REST API but this happens inconsistently 

### Actual behavior
Normally when I scan my NovoPen the treatments are uploaded to Nightscout in <30 seconds via the REST API, however, there is inconsistently in this, sometimes records do not upload to Nightscout via the REST API. The Nightscout instance is a Digital Ocean droplet with firewall rules configured to only allow HTTP and HTTPS traffic through Cloudflare.
https://www.cloudflare.com/en-gb/ips/

I initially thought this could be caused by firewall rules on Cloudflare but there's nothing being blocked as I allow all traffic in the UK to be whitelisted as I have some displays that fetch data from Nightscout and this simply avoids the issue of manually whitelisting other attributes.

### Steps to reproduce the behavior:
For example:
1. Enable the insulin pen scanning like so.
![image](https://github.com/NightscoutFoundation/xDrip/assets/62076535/1cb0084d-da3a-4fb0-ae84-84a2640269bd)
2. Configure the RESTI API in the Cloud Upload section with a valid API token to allow uploading of treatment data.
3. 
![image](https://github.com/NightscoutFoundation/xDrip/assets/62076535/20b2a057-69be-4dd6-89e5-16cc1421cd48)
![image](https://github.com/NightscoutFoundation/xDrip/assets/62076535/2e36e6cd-68fe-4f18-a2c5-ace14cf58967)
4. Tap NovoPen to the back of the phone until the treatments display on xDrip
5. Wait a few minutes at max and see no record being uploaded to my Nightscout instance like so. 
![image](https://github.com/NightscoutFoundation/xDrip/assets/62076535/53c6d927-5f16-4e41-81a8-053f2596729c)
![image](https://github.com/NightscoutFoundation/xDrip/assets/62076535/af38b573-3611-4480-b1a5-891343197f7d)

As you can see from the screenshots the data is not reflecting what's uploaded to Nightscout.

**Comment by Navid200** [comment]
Please let's talk about this as a discussion before turning it into an issue.

**Comment by Navid200** [comment]
Can you check the xDrip logs to see if there is any log related to any failed treatment upload to Nightscout?

**Comment by rossiscool123** [comment]
This is the output from the log when I tried to upload the treatment again after deleting it from xDrip. I assume the "Beyond maximum units value" is the issue but where is this value set? @Navid200 
![image](https://github.com/NightscoutFoundation/xDrip/assets/62076535/2675271b-b4a2-4f78-8c30-e4f5400b1ee8)

**Reply by Navid200** [reply]
It is a constant limit in xDrip.

You have taken 170 units of insulin?  For real?

**Reply by rossiscool123** [reply]
If you're talking daily, no. My Nightscout data shows roughly 20-35 units a day. I don't know where 170 has come from.
@Navid200

**Reply by rossiscool123** [reply]
Is this 170 units in one reading or 170 units in a 24 hour period, the pen has a max reading of 99 before it'll display "Error" on the display and a max dose in one go of 30 units. Interesting where this 170 has come from.

**Reply by Navid200** [reply]
No, there is no daily limit in xDrip.
But, when you attempt to upload an insulin dose to Nightscout, there is a limit of 60 units.  This is there to avoid a typo.  I mean there has to be a limit.  You can question why 60 and not 59.  The answer is that there has to be a reasonable limit for any constant in a program.

I don't know where the value is coming from.  But, this is the reason it is not uploaded.

**Reply by rossiscool123** [reply]
I dosed another two units as a test, this uploaded just fine. But the logs say otherwise. This is what I mean by inconsistent. 
![image](https://github.com/NightscoutFoundation/xDrip/assets/62076535/45e085d2-5ae7-4a89-8725-994c61971556)
![image](https://github.com/NightscoutFoundation/xDrip/assets/62076535/5cd2bdca-7f8d-41ba-a204-8bb2835ef58f)

**Comment by rossiscool123** [comment]
So new issue that's coming up.
![image](https://github.com/NightscoutFoundation/xDrip/assets/62076535/bb2677bf-1c5d-4469-9b77-bb0a94a72beb)

**Comment by Navid200** [comment]
Have you seen this?
https://github.com/NightscoutFoundation/xDrip/issues/2074

Is there anything related?
